time.go

```diff
@@ -0,0 +1,120 @@
+package pflag
+
+import (
+	"fmt"
+	"strings"
+	"time"
+)
+
+// TimeValue adapts time.Time for use as a flag.
+type TimeValue struct {
```

Comment:
Should this really be exported?

The approach we're taking in #432 is to rather expose the constructor functions (to ensure that the struct is correctly initialized before usage); I think the best way here would be to keep this unexported and then let a rebase of #432 handle exporting it.